R/metadata_to_coldata.R

```diff
@@ -0,0 +1,59 @@
+#' Add sample metadata to colData of SCE object
+#'
+#' @param sce SingleCellExperiment object to add sample metadata to colData.
+#'   Must contain the `batch_column` as a column in `colData`.
+#' @param batch_column A character value giving the colData column name
+#'  to differentiate originating SingleCellExperiment objects. Often these values
+#'  are unique library IDs. Default value is `library_id`.
+#'
+#' @return SingleCellExperiment object with the contents of `metadata(sce)$sample_metadata`
+#'   added as new columns to the `colData(sce)`.
+#' @export
+#'
+metadata_to_coldata <- function(sce,
+                                batch_column = "library_id") {
```

Comment:
The reason I used `batch_column` is just because we use it in other places in `scpcaTools`. Should I change it just here and in `merge_sce_list`? Or just here? 

Also, I thought we weren't sure how we planned to handle cell hashing data with the AnnData objects. So I hadn't been thinking about those here. But I do like the idea of being able to join by multiple columns if desired.

Reply:
Actually based on the fact that we would be allowing multiple columns, I think I'm going to change the variable name to `join_columns`.